src/Source/FileSource.js

```diff
@@ -155,17 +157,19 @@ class FileSource extends Source {
     }
 
     onLayerAdded(options) {
+        options.in = this;
         super.onLayerAdded(options);
-        let features = this._featuresCaches[options.crsOut].getByArray([0]);
+        let features = this._featuresCaches[options.out.crs].getByArray([0]);
         if (!features) {
-            options.buildExtent = true;
+            options.out.buildExtent = true;
+            options.out.forceExtentCrs = options.out.crs != 'EPSG:4978' ? options.out.crs : this.crs;
```

Comment:
What about `this.crs == 'EPSG:4978'` ?

Reply:
handled: 

is this.crs == 'EPSG:4978'  so buildExtent = false